lib/smart_answer/money.rb

```diff
@@ -32,8 +25,21 @@ def <=>(other)
       end
     end
 
-    def self.valid?(raw_input)
-      raw_input.is_a?(Numeric) || raw_input.is_a?(Money) || raw_input =~ /\A *[0-9]+(\.[0-9]{1,2})? *\z/
+    def self.raw_input_to_number(raw_input)
+      raw_input = raw_input.is_a?(Numeric) ? raw_input : raw_input.to_s.delete(',').gsub(/\s/, '')
+      if raw_input.to_f.infinite?
+        raise InvalidResponse, "Sorry, that number is too big. Please try again.", caller
```

Comment:
I still find it a bit untidy how our validation logic is tangled up in the casting logic.

I basically expect these three separate steps to happen:

- Attempt to coerce the raw input into a number
- Validate the number (and raise if appropriate)
- Cast the number to a `BigDecimal`

I'd expect these three steps to be represented at a top-level function (probably `initialize`), and to each have a single line of code, calling to methods if required. This way you get proper separation of concerns.